comms/src/proto/rpc.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package tari.comms.rpc;
+
+// Message type for all RPC requests
+message RpcRequest {
+    // An identifier that is unique per request per session. This value is not strictly needed as
+    // requests and responses alternate on the protocol level without the need for storing and matching
+    // request IDs. However, this value is useful for logging.
+    uint32 request_id = 1;
+    // The method identifier. The matching method for a given value is defined by each service.
+    uint32 method = 2;
+    // Message flags. Currently this is not used for requests.
+    uint32 flags = 3;
+    // The length of time in seconds that a client is willing to wait for a response
+    uint64 deadline = 4;
```

Comment:
Ye it's over the wire - the client tells the server how long they're willing to wait. If the "processing part" (i.e. the part that a dev implements) completes before the duration no problem. Otherwise, the server should give up because the client is not going to care about the response anyway. Obviously there is a guard where if the client sets this below some minimum, the server immediately rejects the request (to prevent the client setting some silly deadline like 10ms) - a deadline would typically be 30 seconds.

Reply:
In reality client waits a little longer than the deadline (deadline + grace period) to account for latency (we wouldn't want a server to send a response just within the deadline, but the client has gone away just before the response can be received)